compiler/damlc/daml-prim-src/GHC/Classes.daml

```diff
@@ -407,46 +419,95 @@ class NumericScale (n : Nat) where
-instance NumericScale 13 where numericScale _ = 13; numericScalePrivate _ = ()
-instance NumericScale 14 where numericScale _ = 14; numericScalePrivate _ = ()
-instance NumericScale 15 where numericScale _ = 15; numericScalePrivate _ = ()
-instance NumericScale 16 where numericScale _ = 16; numericScalePrivate _ = ()
-instance NumericScale 17 where numericScale _ = 17; numericScalePrivate _ = ()
-instance NumericScale 18 where numericScale _ = 18; numericScalePrivate _ = ()
-instance NumericScale 19 where numericScale _ = 19; numericScalePrivate _ = ()
-instance NumericScale 20 where numericScale _ = 20; numericScalePrivate _ = ()
-instance NumericScale 21 where numericScale _ = 21; numericScalePrivate _ = ()
-instance NumericScale 22 where numericScale _ = 22; numericScalePrivate _ = ()
-instance NumericScale 23 where numericScale _ = 23; numericScalePrivate _ = ()
-instance NumericScale 24 where numericScale _ = 24; numericScalePrivate _ = ()
-instance NumericScale 25 where numericScale _ = 25; numericScalePrivate _ = ()
-instance NumericScale 26 where numericScale _ = 26; numericScalePrivate _ = ()
-instance NumericScale 27 where numericScale _ = 27; numericScalePrivate _ = ()
-instance NumericScale 28 where numericScale _ = 28; numericScalePrivate _ = ()
-instance NumericScale 29 where numericScale _ = 29; numericScalePrivate _ = ()
-instance NumericScale 30 where numericScale _ = 30; numericScalePrivate _ = ()
-instance NumericScale 31 where numericScale _ = 31; numericScalePrivate _ = ()
-instance NumericScale 32 where numericScale _ = 32; numericScalePrivate _ = ()
-instance NumericScale 33 where numericScale _ = 33; numericScalePrivate _ = ()
-instance NumericScale 34 where numericScale _ = 34; numericScalePrivate _ = ()
-instance NumericScale 35 where numericScale _ = 35; numericScalePrivate _ = ()
-instance NumericScale 36 where numericScale _ = 36; numericScalePrivate _ = ()
-instance NumericScale 37 where numericScale _ = 37; numericScalePrivate _ = ()
+    -- | HIDE
+    oneNumericPrivate: Numeric n
+    -- | HIDE
+    addNumericBuiltin: Numeric n -> Numeric n -> Numeric n
+    -- | HIDE
+    subNumericBuiltin: Numeric n -> Numeric n -> Numeric n
+    -- | HIDE
+    mulNumericBuiltin: forall n1 n2. Numeric n1 -> Numeric n2 -> Numeric n
+    -- | HIDE
+    divNumericBuiltin: forall n1 n2. Numeric n1 -> Numeric n2 -> Numeric n
+    -- | HIDE
+    castNumericBuiltin: forall n1. Numeric n1 -> Numeric n
+    -- | HIDE
+    shiftNumericBuiltin: forall n1. Numeric n1 -> Numeric n
+    -- | HIDE
+    roundNumericBuiltin: Int -> Numeric n -> Numeric n
+    -- | HIDE
+    intToNumericBuiltin: Int -> Numeric n
+--    -- | HIDE
+--    textToNumericBuiltin: Text -> Optional (Numeric n)
+    -- | HIDE
+#ifdef DAML_BIGNUMERIC
+    bigNumericToNumericBuiltin: BigNumeric -> Numeric n
+#endif
 
+    addNumericBuiltin = primitive @"BEAddNumeric"
+    subNumericBuiltin = primitive @"BEAddNumeric"
+    roundNumericBuiltin = primitive @"BERoundNumeric"
```

Comment:
We put those 3 builtin together the one that require a witness for the sake of consistency